### ISSUE TITLE
Hover's float window size problem

### ISSUE THREAD

**Issue by saibing** [onivim/oni#1237]
The oni hover's window:

![oni](https://user-images.githubusercontent.com/5676390/34618702-8bfba1ee-f27a-11e7-9705-ad57c533068a.png)

The vscode hover's window:
![vscode](https://user-images.githubusercontent.com/5676390/34618719-9bb18784-f27a-11e7-9f04-01c343691599.png)

I feeling that vscode's float window is better

**Comment by bryphe** [comment]
Yep, the clipping on the bottom is rough. And we're not properly rendering the `markdown` as we should (#1202 is related to this). Thanks for reporting it + the screenshot, @saibing !

**Comment by Akin909** [comment]
@bryphe regarding this issue I had a look at adding some `margin-bottom` to the quick info which definitely looks nicer to me, and reducing its `max-width` so it doesn't stretch sideways but more vertically. Whilst I was at it I tried using `marked` to render the quickinfo. As far as I can tell from the LSP's typings and docs [here](https://microsoft.github.io/language-server-protocol/specification#textDocument_hover) the hover info is essentially always some form of markdown or as they call it `markup text`, however I noticed you had a todo in the code re how to handle the syntax highlighting, I was going to try and use marked to render it based on the language passed in from the lsp if one is available otherwise I was considering a fallback of the buffers 'filetype'

**Comment by Akin909** [comment]
Tried using `highlight.js` in combination with `marked` which results in:

<img width="500" alt="screen shot 2018-01-07 at 21 31 08" src="https://user-images.githubusercontent.com/22454918/34654449-9f19d730-f3f3-11e7-8969-a2f26137c301.png">

<img width="472" alt="screen shot 2018-01-07 at 21 28 48" src="https://user-images.githubusercontent.com/22454918/34654451-ac505f5a-f3f3-11e7-9b4f-06c0bf6d9d5d.png">

Although think I might have gone off piste as I'm not sure what you might have intended for the highlighting
